### PR TITLE
Get rid of exposed domain separation tags and use dedicated functions instead

### LINKED ISSUE #28: Are NON_INTERACTIVE and X_COORDINATE constants necessary?
They are used to build digests among with several other values, e.g.
```rust
let d = ScalarDigest::new()
    .chain_points(&[precursor, pub_key, dh_point])
    .chain_bytes(NON_INTERACTIVE)
    .finalize();
```
Do we really need to add them? Aren't the values themselves enough?

### PR DESCRIPTION
**Pull request by fjarri**
Fixes #28

As we have discussed, we need to keep the DSTs, but they're better off hidden in the respective functions.

What do you think, keep strings as tags, or use single bytes to make it a little more portable?

### CODE REVIEW

File: umbral-pre/src/hashing_ds.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    pubkey: &CurvePoint,
+    dh_point: &CurvePoint,
+) -> CurveScalar {
+    ScalarDigest::new()
+        .chain_bytes(b"SHARED_SECRET")
+        .chain_point(precursor)
+        .chain_point(pubkey)
+        .chain_point(dh_point)
+        .finalize()
+}
```
Comment (tuxxy) [inline umbral-pre/src/hashing_ds.rs:33]: Looks great.

File: umbral-pre/src/capsule.rs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 use crate::capsule_frag::CapsuleFrag;
-use crate::constants::{NON_INTERACTIVE, X_COORDINATE};
```
Comment (cygnusv) [inline umbral-pre/src/capsule.rs:2]: what was I thinking 🙈

**Review by cygnusv — COMMENTED** [review]
I think this PR will benefit from the discussion on https://github.com/nucypher/rust-umbral/issues/2#issuecomment-789688257

**Comment by fjarri** [comment]
> I think this PR will benefit from the discussion on #2 (comment)

I think that discussion is kind of orthogonal to this PR, is it not? This is just code quality changes. One thing though that could be done is making the DST a parameter in the `ScalarDigest` constructor (leaving it to handle it however it sees fit, if we change the algorithm later). E.g.:
```
    ScalarDigest::new_with_dst(b"SHARED_SECRET")
        .chain_point(precursor)
        .chain_point(pubkey)
        .chain_point(dh_point)
        .finalize()
```

**Pull request merged**